Vapes are coming to SS13

Cuboos opened the pull request:
Bringing the Vape Nation to Space Station 13! Adds new E-cigarettes (other wise known as vapes) to the Cigarette vending machines as contraband. You fill them with reagents instead of dried plants. You can modify them by opening them with a screw driver and increasing the voltage with a Multitool. This creates a small smoke puff with what ever reagent is inside, it also uses reagents faster. You can also maximize the voltage using an emag, which will fill the room with reagent smoke and point you out as a filthy traitor -- useful to spreading meds to an entire room (or poison). Has a 15 percent chance to explode while in use after being emagged, which stuns for a few minutes and deals some burn damage to the mob's head. Will explode and gib you if filled with plasma or will light you on fire if filled with welding fuel. 

Fun uses
- Fill with Leporazine and Salbutamol for ghetto EVA
- Fill with Omnizine for constant healing
- Fill with meth and be a pain in the ass for security. 
- Fill with mannitol as a quick way to cure brain damage
- Fill with Synaptizine and never be bothered by BZ or hallucination sting ever again. 
- Fill with Bath Salts and never get stunned again (also die)
- Fill with welding fuel and leave it around for some poor assistant to find.
- Fill with plasma and put it in the mouth of your assassination target (be sure to run)
- Emag and fill with virus culture, spread nurgle's plague. 
- Mutli-tool it and be a tool your self. 

Added a new explosion sound, a small "pop" bang. Intended for when the vape blows up after being emagged, may also be useful for other things, feel free to use (preview https://dl.dropboxusercontent.com/u/9654207/pop_expl.ogg)

Added a Vape Naysh shirt as contraband to the clothing vendor. Look like the tool that you are.

Cuboos committed:
Vapes are coming to SS13

Iamgoofball commented:
is this a port of my old vape naysh PR or not

Iamgoofball commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+				reagents.trans_to(C, REAGENTS_METABOLISM)
+				if(reagents.get_reagent_amount("welding_fuel"))
+					//HOT STUFF
+					C.fire_stacks = 2;
```
;

optimumtact replied:
stray colon

Cuboos replied:
Sorry... old habit...

Iamgoofball replied:
*semicolon

Cuboos commented:
I didn't even know you made. This was entirely made by scratch from me, with some advice from Coderbus.

Iamgoofball commented:
go green

Iamgoofball reviewed (approved):
![0r80idg](https://cloud.githubusercontent.com/assets/4081722/18734431/3658f3fa-8029-11e6-842e-2b8608426dc9.png)

Cuboos committed:
Update cigs_lighters.dm

Cuboos commented:
Go away goofball

ShadowDeath6 commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -321,6 +321,12 @@
 
+/obj/item/clothing/suit/vapeshirt //wearing this is asking to get beat.
+	name = "Vape Naysh shirt"
+	desc = "A cheap white T-shirt with a big tacky \"VN\" on the front, Why would you wear this unironicly?"
```
unironically

Iamgoofball replied:
intentional, it's a reflection on ethics and irony journalism

Cuboos committed:
fixed the yml

ShadowDeath6 commented:
this better have a big enough capacity to OD people if used with stuff like straight omnizine

Yoshmaster commented:
This looks fun as heck.

Cuboos committed:
Rename example.yml to vapeUpdate.yml

fixed name

Cuboos committed:
hopfully fixed changelog

Cuboos committed:
Merge branch 'VapeNaysh_Yall' of https://github.com/Cuboos/-tg-station into VapeNaysh_Yall

# Conflicts:

Cuboos commented:
GOD FUCKING DAMN IT I HAVE NO IDEA WHAT I'M DOING AND NO ONE IS HELPING

PKPenguin321 commented:
You don't code in the change log you make your changelog with the PR body

See the :CL: things in other PRs for an example of this

Cuboos committed:
oh god i have no idea what i'm doing

Cuboos commented:
I literally have no idea what to do and everyone seems to think i should already understand it.

Signmark commented:
>Fill with Space Drugs and Blood to vape carpets
>Make colorful reagent and bring FRIENDSHIP and TOLERANCE to station 
>Sugar and banana for never-ending fun
why

Cuboos committed:
please work, i am not a good coder

optimumtact committed:
Unfuck the example yml

optimumtact committed:
Merge branch 'VapeNaysh_Yall' of https://github.com/Cuboos/-tg-station into VapeNaysh_Yall

Conflicts:
	html/changelogs/example.yml

Cuboos committed:
Merge pull request #3 from optimumtact/VapeNaysh_Yall

Vape naysh yall

Cuboos commented:
YES, IT WORKS. @optimumtact YOU MY BOO BOO CHILD.

PKPenguin321 commented:
Hooray! :clap:

phil235 commented:
This barely use cigarette code at all, and most cigarette procs need to be overriden with an empty proc to be consistent. There is no reason to make this a child of cigarette.

phil235 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+			START_PROCESSING(SSobj, src)
+			light("<span class='notice'>[user] starts puffing on that fine vape</span>")
+		else //it will not start if the vape is opened.
+			user << "<span class='warning'>You need to close the cap first!</span>"
```
..()

phil235 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+	if(C.get_item_by_slot(slot_wear_mask) == src)
+		lit = 0
+		reagents.set_reacting(FALSE)
+		STOP_PROCESSING(SSobj, src)
```
I'm not sure if this needs `..()`.

phil235 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+
+	if(super && vapetime > 3)//Time to start puffing those fat vapes, yo.
+		var/datum/effect_system/smoke_spread/chem/s = new
+		s.set_up(reagents, 1, loc)
```
I have reservations about how OP this is (smoke clouds multiply the initial reagents), so just standing in your own smoke cloud could be extremely effective. Also, the cloud lasts a bit too long to be a good pick for vaping puffs.

AdamElTablawy replied:
You could easily OD yourself as well, not to mention it'd probably be more effective and have less risk of outing yourself/healing others just to use a syringe or patch or something.

The risk mainly exists in toxin clouds like clouds of initro or plasma. Swallow some antitox and epiphenerine and release smog clouds of initro in escape subtly, 10+ kills easy if people aren't smart enough to epipen themselves.

Plasma = plasma flood because plasma smoke is robust AF.

Cuboos replied:
putting plasma in the vape will cause it to instantly explode, it won't get far enough to cause a flood.

phil235 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+					user << "<span class='notice'>You decrease the voltage in the [src]</span>"
+					add_overlay(image(icon, "vapeopen_low"))
+				else
+					user << "<span class='notice'>The [name] can't be modified!</span>"
```
else
return ..()

lordpidey commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+				user << "<span class='notice'>You increase the voltage in the [src]</span>"
+				add_overlay(image(icon, "vapeopen_med"))
+			else
+				if(!emagged)
```
This will never be false.  The else statement for this if will never be run.

AdamElTablawy replied:
Basically, remove the else.

Cuboos replied:
The else is there to keep it from being lowered if it's already been emagged. Basically if you emag it, it's broken.

lordpidey replied:
the code boils down to
`if(screw && !emagged)`
	`	if(!emagged)`

The else will LITERALLY never be called.

optimumtact replied:
yes

AdamElTablawy commented:
Yet more emag functionality. It'd be better if it was like, if multitooled/EMPd rather than emag, and decrease the potency of the smoke cloud altogether.

Also again, smoke clouds tend to multiply the reagents. A plasma vape could end up flooding a large part of primary with plasma.

Iamgoofball commented:
@Adam fuck off with your bitching about emags

Iamgoofball commented:
I did it from my phone don't judge

Cuboos committed:
Update cigs_lighters.dm

Vapes are no longer a child of cigarette. All references and inherited code from cigarettes have been removed.

Cuboos committed:
Update vending.dm

Added the updated vape to the cigarette contraband list.

optimumtact commented:
I dont' think this should send reagents to other people  in the smoke cloud, unless it's emagged.

It should just deliver them to you and make effect smoke

Cuboos commented:
@phil235 The fixes have been implemented. Has it been 24 hours yet?

Cuboos commented:
@optimumtact It really won't... the non-emagged version only makes on single puff cloud, which can be avoided.

Ninja137 commented:
Has a 15 percent chance to explode while in use after being emagged, which stuns for a few minutes and deals some burn damage to the mob's head.

FOR A FEW MINUTES
WHAT

Ninja137 commented:
Why the hell would you make it explode when emagged, too? You're specifically trying to give it a traitor function that's useful, but then giving a % chance to assfuck whoever tries to actually use said function, when it's not even going to be all that useful in the first place.

Cuboos commented:
@Ninja137 This isn't a very serious PR... but it can be very griefy if used right. Also i miss spoke, not a few minutes, a few seconds.

Cuboos commented:
Well we'll do the minor tweaks later then. We'll see how everyone reacts to them first, maybe get some suggestions. We don't even know if they'll be accepted by the players.

Ninja137 commented:
Goofball was accepted by the players, so you've got pretty good chances.

Cuboos commented:
I don't know... people are really salty about Freon right now...

Ninja137 commented:
Those people are memes, fuck them.

Cuboos commented:
I'm not a meme and i'm kind of annoyed with Freon... I want freon... but like freon to work and not kill the server.

Cuboos committed:
Update cigs_lighters.dm

After another looking over, i have concluded you were right again and i am an idiot. BRB squeezing orange juice in my eyes.

Cuboos commented:
Alright i think i've managed to successfully address all code comments.

phil235 commented on code/game/objects/items/weapons/cigs_lighters.dm:
```diff
@@ -573,3 +573,177 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+		item_state = "[param_color]_vape"
+
+/obj/item/clothing/mask/vape/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/reagent_containers))
```
attack a vape with a burger: "You add the contents of the burger to the vape"

nullbear replied:
yeah, you need to do a 'is sealed' check whatever that is.

Supermichael777 replied:
if(O.spillable) can't do it in the same if because runtimes.

Cuboos commented:
Food is considered a reagent container?

Cuboos committed:
Update cigs_lighters.dm

They can now only be filled by beakers, drinking glasses and medicine bottles.

Cuboos commented:
Anything else? Am i done now? can we merge already?

Chocobro commented:
I was asked to merge this
give me the power of greyskull

Cuboos commented:
Do it.

Cuboos commented:
Beep Boop

Iamgoofball commented:
we get it, you vape

phil235 commented:
There may be some issues with code that checks for cigarette to see if your mask is a  smokable.

Cuboos commented:
@phil235 does that mean i should have kept it a child of cigarette?

phil235 commented:
It means cigarette code is bad to begin with tbh. Maybe we're going to need to make a new /clothing/mask/smokable , with smokable/cigarette and smokable/vape. Look in the code for places checking for the cigarette typepath and decide if the code not catching your vape is a bad thing.

Cuboos commented:
It works fine when i was testing it.
